### PR TITLE
[civ2][flakiness/4] add a script to run tests

### PR DESCRIPTION
**Pull request by can-anyscale**
Add a wrapper script for running unit tests on CI. Turn that into a package and add bazel test run for it.

Test:
- CI

### CODE REVIEW

File: ci/v2/test/t.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        .split("\n")
+    )
+    with open(f"{CURRENT_DIR}/{team}.flaky", "rb") as f:
+        flaky_tests = f.read().decode("utf-8").split("\n")
+
+    return [test for test in test_targets if test not in flaky_tests]
+
+
+if __name__ == "__main__":
+    main()
```
Comment (aslonnie) [inline ci/v2/test/t.py:78]: could you give this script a more meaningful name rather than just `t.py`?

File: ci/v2/test/core.flaky [hunk 2]
```diff
@@ -4,6 +4,7 @@
 //python/ray/tests:test_placement_group_5
 //python/ray/tests:test_runtime_env_2
 //python/ray/tests:test_gcs_fault_tolerance
+//python/ray/tests:test_gcs_ha_e2e
```
Comment (aslonnie) [inline ci/v2/test/core.flaky:7]: what is this line for?

File: ci/v2/test/t.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+)
+@click.option(
+    "--shard",
+    default=0,
+    type=int,
+    help=("Index of the concurrent shard to run."),
+)
+def main(targets: str, team: str, concurrency: int, shard: int) -> None:
+    test_targets = get_test_targets(targets, team)
+    run_tests(np.array_split(test_targets, concurrency)[shard].tolist())
```
Comment (aslonnie) [inline ci/v2/test/t.py:28]: - maybe don't use numpy? feels that it is too heavy to import numpy just for splitting an array.
- what will happen when the array is empty or has fewer than the number of shards?
- could you add a unit test?

File: ci/v2/test/core.flaky (hunk 2 above)
Comment (can-anyscale) [reply ci/v2/test/core.flaky:7]: this test needs a special test environment setup so I keep them in v1 for now

File: ci/ray_ci/test/core.tests.yml [hunk 4]
```diff
@@ -5,6 +5,7 @@ flaky_tests:
   - //python/ray/tests:test_placement_group_5
   - //python/ray/tests:test_runtime_env_2
   - //python/ray/tests:test_gcs_fault_tolerance
+  - //python/ray/tests:test_gcs_ha_e2e
```
Comment (can-anyscale) [inline ci/ray_ci/test/core.tests.yml:8]: This test needs a different test setup environment so I keep them in v1 for now

**Comment by can-anyscale** [comment]
Address @aslonnie's comments. 
- Add unit-tests
- Rename script to runner.py
- I keep using numpy. Splitting array into equal chunks and cover edge cases turn out to be complicated math. Also numpy is declared in requirements.txt so the oss_ci_build is pre-installed with it anyway.

**Review by aslonnie — COMMENTED** [review]
this is draft now?

**Comment by can-anyscale** [comment]
@aslonnie yeah a draft, i failed to resolve the bazel dependency thing; probably will need your help tomorrow

**Comment by can-anyscale** [comment]
Rename BUILD to BUILD.bazel and it works on window now

**Comment by aslonnie** [comment]
> Rename BUILD to BUILD.bazel and it works on window now

we probably should just rename all the `BUILD` files into `BUILD.bazel` for consistency..

File: ci/setup.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        include=[
+            package for package in find_packages() if package.startswith("ray_ci")
+        ],
+    ),
+    version="0.0.1",
+    author="Ray Team",
+    description="The Ray OSS ci testing package",
+    url="https://github.com/ray-project/ray",
+    install_requires=["numpy"],
+)
```
Comment (aslonnie) [inline ci/setup.py:16]: how about I convert this into hermetic bazel for you, so that you do not need this `setup.py` thing?

Comment (can-anyscale) [reply ci/setup.py:16]: Absolutely ;)

**Review by aslonnie — COMMENTED** [review]
will wait for the merge and rebase of the bazel hermetic python thing.

File: ci/ray_ci/test_runner.py [hunk 6]
```diff
@@ -27,35 +27,19 @@ def _mock_check_call(input: List[str]) -> None:
...
-  - //python/ray/tests:test_threaded_actor
-  - //python/ray/tests:test_unhandled_error
-"""
-
-
 def test_get_test_targets() -> None:
+    _TEST_YAML = """
+    flaky_tests:
+        - //python/ray/tests:flaky_test_01
+    """
```
Comment (aslonnie) [inline ci/ray_ci/test_runner.py:41]: the inserted leading tabs feels a bit weird, but probably fine anyways.

maybe just change this to `"flaky_tests: [ //python/ray/tests:flaky_test_01 ]"`, which is a single line string.

**Review by aslonnie — APPROVED** [review]

File: ci/ray_ci/test_runner.py (hunk 6 above)
Comment (can-anyscale) [reply ci/ray_ci/test_runner.py]: fixed!

**Pull request merged**